treescript/src/treescript/gecko/l10n.py

```diff
@@ -232,3 +275,58 @@ async def l10n_bump(config, task, repo_path):
         await vcs.commit(config, repo_path, message)
         changes += 1
     return changes
+
+
+# l10n_bump_github {{{1
+async def l10n_bump_github(config: dict, task: dict, repo_path: str) -> int:
+    """Perform a l10n revision bump.
+
+    This function takes its inputs from task by using the ``get_l10n_bump_info``
+    function from treescript.task. It then calculates the locales, the platforms
+    for each locale, and the locale revision for each locale.
+
+    Args:
+        config (dict): the running config
+        task (dict): the running task
+        repo_path (str): the source directory
+
+    Raises:
+        TaskVerificationError: if a file specified is not allowed, or
+                               if the file is not in the target repository.
+
+    Returns:
+        int: non-zero if there are any changes.
+
+    """
+    log.info("Preparing to bump l10n changesets.")
+
+    ignore_closed_tree = get_ignore_closed_tree(task)
+    if not ignore_closed_tree and not task["payload"].get("dry_run", False):
+        if not await check_treestatus(config, task):
+            log.info("Treestatus is closed; skipping l10n bump.")
+            return 0
+
+    dontbuild = get_dontbuild(task)
+    l10n_bump_info = get_l10n_bump_info(task)
+    changes = 0
+
+    for bump_config in l10n_bump_info:
+        l10n_repo_url = bump_config.get("l10n_repo_url")
+        if not l10n_repo_url:
+            raise Exception("fail")
+
+        owner, repo = extract_github_repo_owner_and_name(l10n_repo_url)
+        async with GithubClient(config, owner, repo) as client:
+            path = os.path.join(repo_path, bump_config["path"])
+            old_contents = load_json_or_yaml(path, is_path=True)
+            # find latest revision in l10n_repo_url
+            new_contents = await build_revision_dict_github(client, bump_config, repo_path)
+            if old_contents == new_contents:
+                continue
+            with open(path, "w") as fh:
+                fh.write(json.dumps(new_contents, sort_keys=True, indent=4, separators=(",", ": ")))
+            locale_map = build_locale_map(old_contents, new_contents)
+            message = build_commit_message(bump_config["name"], locale_map, dontbuild=dontbuild, ignore_closed_tree=ignore_closed_tree)
+            await vcs.commit(config, repo_path, message)
```

Comment:
Maybe I'm misunderstanding something, but doesn't this only work for Mercurial? For Github I think you need to use the client + GraphQL API to commit. There's no checkout to work with afaict.

There's an example of this here:
https://github.com/mozilla-releng/scriptworker-scripts/blob/da0149e6de61cbe5e6a00723b5b9d4867a61afe2/treescript/src/treescript/github/versionmanip.py#L99

At a high level the above is:
1. Calling `client.get_files(...)` on all files that need to be modified.
2. Computing the diff between the old and new versions of the files.
3. Calling `client.commit` with the diff info.

It's a bit more complicated to implement, but it's kind of nice to not need to deal with cloning or anything.

Reply:
We're not committing anything to Github here. The new repository is where the translated strings are held, and we need to find the latest revision of them when we run l10n bumper or merge automation, and commit that back to Gecko.

When we migrated Gecko to GitHub I think we'll want to do what you suggest to avoid needing a checkout  for these tasks though?

Reply:
Thanks, that's what I was missing